src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+    def generate_ij_loop(
+        self, block_extents: dict[int, Extent], execution: HorizontalExecution
+    ) -> Generator:
+        extents = block_extents[id(execution)]
+        i_loop = f"for i in range(i_0 + {extents[0][0]} , i_size + {extents[0][1]}):"
+        self.body.append(i_loop)
+        self.body.indent()
+        j_loop = f"for j in range(j_0 + {extents[1][0]} , j_size + {extents[1][1]}):"
+        self.body.append(j_loop)
+        self.body.indent()
+        yield
+        self.body.dedent()
+        self.body.dedent()
+
+    def visit_While(self, while_node: While, **_) -> None:
+        while_condition = self.visit(while_node.cond)
+        while_code = f"while {while_condition}:"
+        self.body.append(while_code)
+        self.body.indent()
+        for statement in while_node.body:
+            self.visit(statement)
+        self.body.dedent()
+
+    def visit_FieldDecl(self, field_decl: FieldDecl, **_) -> str:
+        return str(field_decl.name)
+
+    def visit_AxisBound(self, axis_bound: AxisBound, **kwargs):
+        if axis_bound.level == LevelMarker.START:
+            return f"{kwargs['var']}_0 + {axis_bound.offset}"
+        if axis_bound.level == LevelMarker.END:
+            return f"{kwargs['var']}_size + {axis_bound.offset}"
+
+    def visit_Interval(self, interval: Interval, **kwargs):
+        if kwargs["direction"] == LoopOrder.BACKWARD:
+            return ",".join(
+                [
+                    self.visit(interval.end, **kwargs) + "- 1",
+                    self.visit(interval.start, **kwargs) + "- 1",
+                ]
+            )
+        else:
+            return ",".join(
+                [self.visit(interval.start, **kwargs), self.visit(interval.end, **kwargs)]
+            )
+
+    def visit_Temporary(self, temporary_declaration: Temporary, **kwargs) -> str:
+        field_extents = kwargs["field_extents"]
+        local_field_extent = field_extents[temporary_declaration.name]
+        i_padding: int = local_field_extent[0][1] - local_field_extent[0][0]
+        j_padding: int = local_field_extent[1][1] - local_field_extent[1][0]
+        shape: list[str] = [f"i_size + {i_padding}", f"j_size + {j_padding}", "k_size"]
+        data_dimensions: list[str] = [str(dim) for dim in temporary_declaration.data_dims]
+        shape = shape + data_dimensions
+        shape_decl = ", ".join(shape)
+        dtype: str = self.visit(temporary_declaration.dtype)
+        field_offset = tuple(-ext[0] for ext in local_field_extent)
+        offset = [str(off) for off in field_offset] + ["0"] * (
+            1 + len(temporary_declaration.data_dims)
+        )
+        return f"{temporary_declaration.name} = Field.empty(({shape_decl}), {dtype}, ({', '.join(offset)}))"
+
+    def visit_DataType(self, data_type: DataType, **_) -> str:
+        if data_type not in {DataType.BOOL}:
+            return f"np.{data_type.name.lower()}"
+        else:
+            return data_type.name.lower()
```

Comment:
```suggestion
    def visit_DataType(self, data_type: DataType, **_) -> str:
        if data_type in {DataType.BOOL}:
            return data_type.name.lower()

        return f"np.{data_type.name.lower()}"
```

Handle special case first. Avoid `else` after `return` in `if` branch.